### PR TITLE
plasma5: 5.21.5 -> 5.22

### PR DESCRIPTION
**Pull request by CertainLach**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127389` at 4f0a6948 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>34 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/2c0d3421cf3dd3774b834b1308038f2f">libsForQt512.karchive</a></li>
    <li><a href="https://gist.github.com/f3b747b8ccd617f2c593c4913fd1bb8b">libsForQt512.kdecoration</a></li>
    <li><a href="https://gist.github.com/e02c2529f7c1b00d2ee783612789cc0d">libsForQt512.ki18n</a></li>
    <li><a href="https://gist.github.com/8fec720eefe5bb5217e567bd2a74f62a">libsForQt514.karchive</a></li>
    <li><a href="https://gist.github.com/897d544759a02ba56941a765dd9fcb6e">libsForQt514.kauth</a></li>
    <li><a href="https://gist.github.com/9a17a80ed03c3ec4e9b32cb74de81c23">libsForQt514.kcodecs</a></li>
    <li><a href="https://gist.github.com/7b0e363dfd3c3578f30fad93e28a7c97">libsForQt514.kconfig</a></li>
    <li><a href="https://gist.github.com/3c7641a02b179aa711f269630e0d782b">libsForQt514.kconfigwidgets</a></li>
    <li><a href="https://gist.github.com/1743dd9754656ab6def1293d3c909ad7">libsForQt514.kcoreaddons</a></li>
    <li><a href="https://gist.github.com/df314be527bc0886e783dec946dc73d9">libsForQt514.kcrash</a></li>
    <li><a href="https://gist.github.com/286ff55fc8e456113eaea3fb93227b1d">libsForQt514.kdbusaddons</a></li>
    <li><a href="https://gist.github.com/25fd9a840dd417aa3d6a13045ed94221">libsForQt514.kdecoration</a></li>
    <li><a href="https://gist.github.com/9512634bf1abfe3780ad1744438ea32d">libsForQt514.kdoctools</a></li>
    <li><a href="https://gist.github.com/1e1bd2316e2b5ae41b6327be3b9f8b04">libsForQt514.kgamma5</a></li>
    <li><a href="https://gist.github.com/74e589cf183682401cb99f2bdc1dfdc1">libsForQt514.kguiaddons</a></li>
    <li><a href="https://gist.github.com/a2c57cbbd8c34376f20bd3b4d2f55131">libsForQt514.ki18n</a></li>
    <li><a href="https://gist.github.com/78e9912e55d52e236fa5602379e14fc8">libsForQt514.kiconthemes</a></li>
    <li><a href="https://gist.github.com/b43a70e7074fdd78e262d67785bc4e43">libsForQt514.kidletime</a></li>
    <li><a href="https://gist.github.com/57531c882104374ba689b01a2bced4e3">libsForQt514.kirigami2</a></li>
    <li><a href="https://gist.github.com/4a029111d24e16a506be2453e5230756">libsForQt514.kitemviews</a></li>
    <li><a href="https://gist.github.com/60e1a8afc3b4f69fed7693c293dd62b0">libsForQt514.knotifications</a></li>
    <li><a href="https://gist.github.com/343d8b3ef44a62941d5065884490e239">libsForQt514.kpty</a></li>
    <li><a href="https://gist.github.com/6b0c5cbdbac65c493a80c9317cb6e229">libsForQt514.kservice</a></li>
    <li><a href="https://gist.github.com/838d1cc6a3bd51b9dcee6da74d284550">libsForQt514.ksshaskpass</a></li>
    <li><a href="https://gist.github.com/91eb99d90142ba3f57d10e3cdfe4ffc5">libsForQt514.kwallet</a></li>
    <li><a href="https://gist.github.com/64f4ad0468ed204fe814b17b932306a8">libsForQt514.kwallet-pam</a></li>
    <li><a href="https://gist.github.com/0d5bf6ccf96b7dde2e1448b01e1ae0ad">libsForQt514.kwayland</a></li>
    <li><a href="https://gist.github.com/d64021d973496646d5078f5884776b4b">libsForQt514.kwayland-integration</a></li>
    <li><a href="https://gist.github.com/a4ce1b7eb0bae2bfa2bd91c2a1bae873">libsForQt514.kwidgetsaddons</a></li>
    <li><a href="https://gist.github.com/0a787fc7d7d750aa105e2b04ef26c0c7">libsForQt514.kwindowsystem</a></li>
    <li><a href="https://gist.github.com/889ae4a9946eb68f579d743aed7d6714">libsForQt514.kwrited</a></li>
    <li><a href="https://gist.github.com/c306c7ac9b22cc6a8f8519faf8346cdc">libsForQt514.layer-shell-qt</a></li>
    <li><a href="https://gist.github.com/316e01821a3e3a301d5776dc0be4420d">libsForQt514.polkit-kde-agent</a></li>
    <li><a href="https://gist.github.com/a77602f24229c0c24f70f4c76fe6a31e">libsForQt514.qqc2-breeze-style</a></li>
  </ul>
</details>
<details>
  <summary>997 packages skipped due to time constraints:</summary>
  <ul>
    <li>ajour</li>
    <li>amarok (amarok-kf5)</li>
    <li>angelfish</li>
    <li>antimicroX</li>
    <li>bcompare</li>
    <li>beauty-line-icon-theme</li>
    <li>calamares</li>
    <li>calligra</li>
    <li>colord-kde</li>
    <li>corectrl</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>51 packages built successfully:</summary>
  <ul>
    <li>extra-cmake-modules (libsForQt5.extra-cmake-modules ,libsForQt515.extra-cmake-modules ,plasma5Packages.extra-cmake-modules)</li>
    <li>libsForQt5.breeze-grub (libsForQt515.breeze-grub ,plasma5Packages.breeze-grub)</li>
    <li>libsForQt5.breeze-icons (libsForQt515.breeze-icons ,plasma5Packages.breeze-icons)</li>
    <li>libsForQt5.breeze-plymouth (libsForQt515.breeze-plymouth ,plasma5Packages.breeze-plymouth)</li>
    <li>libsForQt5.karchive (libsForQt515.karchive ,plasma5Packages.karchive)</li>
    <li>libsForQt5.kauth (libsForQt515.kauth ,plasma5Packages.kauth)</li>
    <li>libsForQt5.kcodecs (libsForQt515.kcodecs ,plasma5Packages.kcodecs)</li>
    <li>libsForQt5.kconfig (libsForQt515.kconfig ,plasma5Packages.kconfig)</li>
    <li>libsForQt5.kconfigwidgets (libsForQt515.kconfigwidgets ,plasma5Packages.kconfigwidgets)</li>
    <li>libsForQt5.kcoreaddons (libsForQt515.kcoreaddons ,plasma5Packages.kcoreaddons)</li>
    <li>libsForQt5.kcrash (libsForQt515.kcrash ,plasma5Packages.kcrash)</li>
    <li>libsForQt5.kdbusaddons (libsForQt515.kdbusaddons ,plasma5Packages.kdbusaddons)</li>
    <li>libsForQt5.kdecoration (libsForQt515.kdecoration ,plasma5Packages.kdecoration)</li>
    <li>libsForQt5.kdoctools (libsForQt515.kdoctools ,plasma5Packages.kdoctools)</li>
    <li>libsForQt5.kgamma5 (libsForQt515.kgamma5 ,plasma5Packages.kgamma5)</li>
    <li>libsForQt5.kguiaddons (libsForQt515.kguiaddons ,plasma5Packages.kguiaddons)</li>
    <li>libsForQt5.ki18n (libsForQt515.ki18n ,plasma5Packages.ki18n)</li>
    <li>libsForQt5.kiconthemes (libsForQt515.kiconthemes ,plasma5Packages.kiconthemes)</li>
    <li>libsForQt5.kidletime (libsForQt515.kidletime ,plasma5Packages.kidletime)</li>
    <li>libsForQt5.kirigami2 (libsForQt515.kirigami2 ,plasma5Packages.kirigami2)</li>
    <li>libsForQt5.kitemviews (libsForQt515.kitemviews ,plasma5Packages.kitemviews)</li>
    <li>libsForQt5.knotifications (libsForQt515.knotifications ,plasma5Packages.knotifications)</li>
    <li>libsForQt5.kpty (libsForQt515.kpty ,plasma5Packages.kpty)</li>
    <li>libsForQt5.kservice (libsForQt515.kservice ,plasma5Packages.kservice)</li>
    <li>libsForQt5.ksshaskpass (libsForQt515.ksshaskpass ,plasma5Packages.ksshaskpass)</li>
    <li>libsForQt5.kwallet (libsForQt515.kwallet ,plasma5Packages.kwallet)</li>
    <li>libsForQt5.kwallet-pam (libsForQt515.kwallet-pam ,plasma5Packages.kwallet-pam)</li>
    <li>libsForQt5.kwayland (libsForQt515.kwayland ,plasma5Packages.kwayland)</li>
    <li>libsForQt5.kwayland-integration (libsForQt515.kwayland-integration ,plasma5Packages.kwayland-integration)</li>
    <li>libsForQt5.kwidgetsaddons (libsForQt515.kwidgetsaddons ,plasma5Packages.kwidgetsaddons)</li>
    <li>libsForQt5.kwindowsystem (libsForQt515.kwindowsystem ,plasma5Packages.kwindowsystem)</li>
    <li>libsForQt5.kwrited (libsForQt515.kwrited ,plasma5Packages.kwrited)</li>
    <li>libsForQt5.layer-shell-qt (libsForQt515.layer-shell-qt ,plasma5Packages.layer-shell-qt)</li>
    <li>libsForQt5.libkscreen (libsForQt515.libkscreen ,plasma5Packages.libkscreen)</li>
    <li>libsForQt5.phonon (libsForQt515.phonon ,plasma5Packages.phonon)</li>
    <li>libsForQt5.plasma-wayland-protocols (libsForQt515.plasma-wayland-protocols ,plasma5Packages.plasma-wayland-protocols)</li>
    <li>libsForQt5.plasma-workspace-wallpapers (libsForQt515.plasma-workspace-wallpapers ,plasma5Packages.plasma-workspace-wallpapers)</li>
    <li>libsForQt5.polkit-kde-agent (libsForQt515.polkit-kde-agent ,plasma5Packages.polkit-kde-agent)</li>
    <li>libsForQt5.qqc2-breeze-style (libsForQt515.qqc2-breeze-style ,plasma5Packages.qqc2-breeze-style)</li>
    <li>libsForQt512.breeze-grub</li>
    <li>libsForQt512.breeze-plymouth</li>
    <li>libsForQt512.extra-cmake-modules</li>
    <li>libsForQt512.plasma-wayland-protocols</li>
    <li>libsForQt512.plasma-workspace-wallpapers</li>
    <li>libsForQt514.breeze-grub</li>
    <li>libsForQt514.breeze-icons</li>
    <li>libsForQt514.breeze-plymouth</li>
    <li>libsForQt514.extra-cmake-modules</li>
    <li>libsForQt514.phonon</li>
    <li>libsForQt514.plasma-wayland-protocols</li>
    <li>libsForQt514.plasma-workspace-wallpapers</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/desktops/plasma-5/discover.nix:37:3:
```
   |
37 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`fdl12` is a deprecated license, please check if project uses `fdl12Plus` or `fdl12Only` and change `meta.license` accordingly.

Near pkgs/desktops/plasma-5/default.nix:95:17:
```
   |
95 |                 license = meta.license or license;
   |                 ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by andrevmatos** [comment]
Plasma 5.22.1 is already out; any change of upgrading directly to it, since it should be just bugfixes?

**Comment by jansol** [comment]
Is KSysGuard completely gone from upstream? If not I'd prefer to keep the package for now and just mark it as deprecated with a note pointing to the new system monitor package so people who for one reason or another insist on still using it can do so for the time being.

**Comment by andrevmatos** [comment]
I'm running this PR now with bump to plasma 5.22.1. So far, an issue I'm seeing is that neither the new System Monitor, nor the previous monitoring widgets I had, can show any meters for CPU, memory, etc... It seems they still depend on maybe some kind of `libksysguard` which apparently was provided by the old `ksysguard` package (at build/runtime?), and since it got completely removed, those remain blank on the current state of this PR.

**Comment by CertainLach** [comment]
`ksysguard` is missing from srcs, it was completly removed from upstream.
`libksysguard` is still present, it is provided as a different package

**Comment by andrevmatos** [comment]
Yes, I know, but since `ksysguard` was removed from `buildInputs`, `system-monitor` app and resources monitoring widgets stopped working, maybe some lib which was provided or pulled by `ksysguard` package either at build or runtime made these functionality stop working

**Comment by CertainLach** [comment]
Seems like those widgets depends on new system-monitor app, and i didn't enabled it instead of ksysguard

**Comment by andrevmatos** [comment]
I have SystemMonitor itself installed explicity (`plasma-systemmonitor` package), but it's also not working, so I think the functionality is broken despite the widgets.

**Comment by CertainLach** [comment]
Have you installed it as `environment.systemPackages`, or via `nix-env`?

**Comment by andrevmatos** [comment]
`systemPackages`

**Comment by andrevmatos** [comment]
Krunner seems to also not be working:
```
plasmashell[224153]: Could not open library 'libkdeinit5_'.
plasmashell[224153]: Cannot load library /run/current-system/sw/lib/libkdeinit5_: (/run/current-system/sw/lib/haswell/liblibkdeinit5_.so: cannot open shared object file: No such file or directory)
```
Notice the `liblibkdeinit5`.. not sure why is that.

**Comment by andrevmatos** [comment]
I think the issue is that plasma has a new package: `ksystemstats`, which replaces `ksysguard` also as dependency of some of the others (e.g. `plasma-systemmonitor`, `plasma-workspace`, etc) to provide resources monitoring integration. I'm trying to build it locally, but there're some weird make errors. So far, `libksysguard`, `libnl` and `lm_sensors` are in its `buildInputs`.

**Comment by CertainLach** [comment]
Built it via
```nix
{
  mkDerivation, lib,
  extra-cmake-modules,
  libksysguard, libnl, lm_sensors, networkmanager-qt
}:

mkDerivation {
  name = "ksystemstats";
  NIX_CFLAGS_COMPILE = [ "-I${libksysguard.bin}/share" ];
  nativeBuildInputs = [ extra-cmake-modules ];
  buildInputs = [ libksysguard libnl lm_sensors networkmanager-qt ];
}
```

Seems like this is a dbus service, which should be started somehow

**Comment by andrevmatos** [comment]
It's working fine here now! Both `plasma-systemmonitor` and monitoring widgets.
I wouldn't use it, but it seems old `ksysguard` is still available on plasma 5.22, not sure why its source didn't get picked up:
https://github.com/archlinux/svntogit-packages/blob/packages/ksysguard/trunk/PKGBUILD

**Comment by CertainLach** [comment]
Because it was moved out of `plasma` subdirectory

**Review by andrevmatos — APPROVED** [review]
I've been testing it for a couple of days now, and everything has been working great so far. Even wayland with nvidia reverse prime works out-of-the-box! It'd be good if `ksysguard` is provided as a package outside of the main plasma5 group. I'm no maintainer, so approving only to provide my experience with the patchset.

**Comment by peterhoeg** [comment]
We are a few kde frameworks versions behind. Any chance you can find the time to do that at the same time to avoid rebuilding everything again?

**Comment by andrevmatos** [comment]
> We are a few kde frameworks versions behind. Any chance you can find the time to do that at the same time to avoid rebuilding everything again?

This PR also bumps frameworks to 5.83, latest stable release.

File: pkgs/development/libraries/kde-frameworks/default.nix [hunk 1]
```diff
@@ -28,7 +28,7 @@ existing packages here and modify it as necessary.
 
 let
 
-  minQtVersion = "5.14";
```
Comment (andrevmatos) [inline pkgs/development/libraries/kde-frameworks/default.nix:31]: It seems `kimageformats` framework has a patch included which fails to apply when rebasing this on top of master because it was already included in `5.83`. Maybe they transformed the warning in an error?

**Comment by peterhoeg** [comment]
The patches to the kwin patches are superfluous. They just add one directory level to the headers.

**Comment by CertainLach** [comment]
Those patches are here because of https://github.com/KDE/kwin/commit/93e0265e4e4501cb5459423892e770029e43edfd

File: pkgs/development/libraries/kde-frameworks/default.nix (hunk 1 above)
Comment (jansol) [reply pkgs/development/libraries/kde-frameworks/default.nix:31]: Yeah 0c556f31caba6f198b2d77f1e4b33d53dd58ee81 needs to be reverted, the patch would technically still apply, but in reverse (and there is a comment on that commit mentioning that the change is in 5.83). The patch applying only if reversed causes nix to panic and bail out.

**Comment by andrevmatos** [comment]
`plasma 5.22.3` is out. @ttuegel or other maintainer, could someone look at this PR, please?

**Comment by CertainLach** [comment]
Rebasing/updating to 5.22.3

**Comment by jakobrs** [comment]
After 1.5 hours of compiling, I've tried this and can confirm that it seems to be working. Things I noticed:

- Konsole looks like it's using the default theme
  - This feels like it happens on every Plasma update anyway
  - Setting `QT_STYLE_OVERRIDE=Breeze` errors with:
    ```
    QApplication: invalid style override 'Breeze' passed, ignoring it.
            Available styles: Breeze, Oxygen, Windows, Fusion
    ```
    all other themes are fine.
    - Could be related to #98268, although only Konsole's affected
- Notifications are still broken on Plasma Wayland
- There is a notification on launch if using runUsingSystemd that says:
  ```
  Unable to create io-slave. klauncher said: Error loading '/nix/store/9aqfx35m0x76f4qa6z6fyil7p5q1hr55-baloo-5.83.0/lib/qt-5.15.2/plugins/kf5/kio/tags.so'.
  ```
  This notifcation also appears in several other places, like the first time you open the "Places" tab in the start menu (per session), and every time you launch Dolphin (and Dolphin doesn't show any files).
  This is printed in the systemd journal:
  ```
  Jul 09 11:24:42 girafarig kded5[2858]: Could not open library '/nix/store/9aqfx35m0x76f4qa6z6fyil7p5q1hr55-baloo-5.83.0/lib/qt-5.15.2/plugins/kf5/kio/tags.so'.
  Jul 09 11:24:42 girafarig kded5[2858]: Cannot load library /nix/store/9aqfx35m0x76f4qa6z6fyil7p5q1hr55-baloo-5.83.0/lib/qt-5.15.2/plugins/kf5/kio/tags.so: (/nix/store/n20vanh45kzh3b6ygkjy4h0qa2w9capd-kio-5.83.0/lib/libKF5KIOCore.so.5: undefined symbol: _ZN4KJob14percentChangedEPS_mNS_14QPrivateSignalE)
  Jul 09 11:24:42 girafarig kded5[2726]: kf.kio.core: couldn't create slave: "klauncher said: Error loading '/nix/store/9aqfx35m0x76f4qa6z6fyil7p5q1hr55-baloo-5.83.0/lib/qt-5.15.2/plugins/kf5/kio/tags.so'."
  Jul 09 11:24:42 girafarig kded5[2726]: kf.kio.core: "Unable to create io-slave. klauncher said: Error loading '/nix/store/9aqfx35m0x76f4qa6z6fyil7p5q1hr55-baloo-5.83.0/lib/qt-5.15.2/plugins/kf5/kio/tags.so'."
  ```
- A string is printed in the middle of the background if using runUsingSystemd
  ```
  Unable to create io-slave: klauncher said: Error loading '/run/current-system/sw/lib/qt-5.15.2/plugins/kf5/kio/desktop.so'.
  ```
  Clearly related to the point above.
- Scaling the screen on Wayland causes everything to be blurry.

No obvious graphical issues introduced. OpenGL works fine.

In conclusion, most things seem to be fine, although there are issues if combined with #117102.

**Comment by CertainLach** [comment]
> Konsole looks like it's using the default theme
Is konsole installed via `nix-env/nix profile`, or added to system packages?

In case of local installation there could be wrapped libraries version mismatch

> undefined symbol: _ZN4KJob14percentChangedEPS_mNS_14QPrivateSignalE
Seems like it has same reason as issue above: library version mismatch

**Comment by jakobrs** [comment]
Konsole is not installed using `nix-env`, no. And the other issue affects, it seems, every part of systemd with file access, including Dolphin, the start menu, and the desktop itself (for displaying desktop icons, presumably), none of which are installed using `nix-env`.

**Comment by andrevmatos** [comment]
`kde-frameworks 5.84` is out now. This is taking way too long to get reviewed.

**Comment by jansol** [comment]
FWIW I've been using this PR for a week now without a hitch. I don't get the error notification nor the blurriness mentioned by @jakobrs either. Applications running in Xwayland are blurry when scaled, but they have always been that way. Wayland-native applications are crisp as a cookie even with scaling.

**Comment by jakobrs** [comment]
Since any potential Wayland issues are strictly speaking part of #117102 and not this PR, I suggest ensuring that this PR works (and it seems it does, at least for the most part) and then worrying about Wayland issues later.

**Comment by matklad** [comment]
Works perfectly for me on X, thanks! [Command Bar](https://pointieststick.com/2021/05/21/this-week-in-kde-kcommandbar-delivers-ludicrous-mode-productivity/) is :fire:

**Comment by jansol** [comment]
There seem to be a decent few positive results from testing both on X11 and Wayland (insofar as that has ever worked). Is there anything actually *preventing* this from getting merged already? This has been hanging for over a month now.

File: pkgs/desktops/plasma-5/ksystemstats.nix [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  mkDerivation, lib,
+  extra-cmake-modules,
+  libksysguard, libnl, lm_sensors, networkmanager-qt
+}:
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/ksystemstats.nix:5]:
```suggestion
{ mkDerivation, lib
, extra-cmake-modules
, libksysguard, libnl, lm_sensors, networkmanager-qt
}:
```

File: pkgs/desktops/plasma-5/layer-shell-qt.nix [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  mkDerivation, lib,
+  extra-cmake-modules,
+  kguiaddons, kidletime, kwayland, kwindowsystem, qtbase, wayland-scanner, wayland, wayland-protocols,
+}:
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/layer-shell-qt.nix:5]:
```suggestion
{ mkDerivation, lib
, extra-cmake-modules
, kguiaddons, kidletime, kwayland, kwindowsystem, qtbase, wayland-scanner, wayland, wayland-protocols
}:
```

File: pkgs/desktops/plasma-5/default.nix [hunk 4]
```diff
@@ -123,12 +123,13 @@ let
       kscreen = callPackage ./kscreen.nix {};
       kscreenlocker = callPackage ./kscreenlocker.nix {};
       ksshaskpass = callPackage ./ksshaskpass.nix {};
-      ksysguard = callPackage ./ksysguard.nix {};
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/default.nix:126]: Should we add a throw alias for ksysguard?

Comment (jansol) [reply pkgs/desktops/plasma-5/default.nix:126]: That would be a reasonable thing to do, yes. It should point people to plasma-systemmonitor.

Comment (andrevmatos) [reply pkgs/desktops/plasma-5/default.nix:126]: `ksysguard` still exists, it's just outside of plasma default distribution, but ideally we should keep it as an optional package on kde-apps for anyone interested.

Comment (CertainLach) [reply pkgs/desktops/plasma-5/default.nix:126]: Best i can do - add throw alias, ksysguard just doesn't build with latest libksysguard, and i am too lazy to package old versions of all of its dependencies :D

File: pkgs/desktops/plasma-5/ksystemstats.nix [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{ mkDerivation, lib
+, extra-cmake-modules
+, libksysguard, libnl, lm_sensors, networkmanager-qt
+}:
+
+mkDerivation {
+  name = "ksystemstats";
+  NIX_CFLAGS_COMPILE = [ "-I${libksysguard.bin}/share" ];
```
Comment (ttuegel) [inline pkgs/desktops/plasma-5/ksystemstats.nix:8]: Just a bit of future-proofing:

```suggestion
  NIX_CFLAGS_COMPILE = [ "-I${lib.getBin libksysguard}/share" ];
```

**Comment by ttuegel** [comment]
This broke some of my global shortcut keys. I'm investigating, to see if we can give a smooth user experience with the update.

**Comment by ttuegel** [comment]
> This broke some of my global shortcut keys.

This was just for KRunner, and it was due to some stale entries in `~/.config/kglobalshortcutsrc`. It was quick to fix from System Settings by deleting the stale entries, but I don't know how we would avoid it in the first place.

**Comment by pasqui23** [comment]
@ttuegel Plasma upgrades often broke my config completely, and if only a few shortcuts are broken then it's fine for me

**Comment by pasqui23** [comment]
Although lately there where no breakages on plasma upgrades, so kudos to the plasma team

**Review by ttuegel — APPROVED** [review]

**Comment by greydot** [comment]
Meanwhile, Plasma 5.22.4 is out. Can we have that release included here?

https://kde.org/announcements/plasma/5/5.22.4/

**Comment by jansol** [comment]
Another week has passed... This isn't exactly encouraging for people considering contributing to the packaging efforts...

I'd be fine with either updating this PR to 5.22.4 or just merging it as is and having another PR for that version bump. The diff should be much smaller and faster to review.

**Comment by acowley** [comment]
Might be worth pointing to the issue on the discourse to see if that can attract the attention of someone with commit permissions. I guess the list of potential reviewers is just out of date.

**Comment by InternetUnexplorer** [comment]
It is a bit discouraging, yeah… :(

Off the top of my head, it might be worth posting in one or both of these two:
- https://discourse.nixos.org/t/prs-already-reviewed/2617 (since it seems like it was already approved by @ttuegel?)
- https://discourse.nixos.org/t/prs-in-distress/3604

There's also https://discourse.nixos.org/t/prs-ready-for-review/3032, but I don't think it's the right place since it seems like this has already been reviewed?

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Comment by happysalada** [comment]
Would anyone be interested in applying for commit rights ?
I think it's clear that the commiters team can't keep up with all the good contributions. Not to mention that lots of contributions require specific knowledge.
This is the issue where commit rights can be requested.
https://github.com/NixOS/nixpkgs/issues/50105

If anything doesn't make sense, feel free to ask.

**Comment by drupol** [comment]
Should we backport this to release-21.05 ?
